tmail-backend/jmap/extensions/src/main/scala/com/linagora/tmail/james/jmap/model/CalendarEventParse.scala

```diff
@@ -37,28 +41,111 @@ object CalendarEventNotParsable {
     CalendarEventNotParsable(notParsable1.value ++ notParsable2.value)
 }
 
+object CalendarTitleField {
+  def from(calendarEvent: VEvent): Option[CalendarTitleField] =
+    Option(calendarEvent.getSummary).map(_.getValue).map(CalendarTitleField(_))
+}
+
 case class CalendarTitleField(value: String) extends AnyVal
+
+object CalendarDescriptionField {
+  def from(calendarEvent: VEvent): Option[CalendarDescriptionField] =
+    Option(calendarEvent.getDescription).map(_.getValue).map(CalendarDescriptionField(_))
+}
+
 case class CalendarDescriptionField(value: String) extends AnyVal
 
+object CalendarLocationField {
+  def from(calendarEvent: VEvent): Option[CalendarLocationField] =
+    Option(calendarEvent.getLocation).map(_.getValue).map(CalendarLocationField(_))
+}
+
+case class CalendarLocationField(value: String) extends AnyVal
+
+object CalendarStartField {
+  def from(calendarEvent: VEvent): Option[CalendarStartField] =
+    Option(calendarEvent.getStartDate)
+      .flatMap(startDate => Option(startDate.getDate)
+        .map(date => CalendarStartField(ZonedDateTime.ofInstant(date.toInstant,
+          Option(startDate.getTimeZone).getOrElse(getTimeZoneAlternative(calendarEvent)).toZoneId))))
+
+  def getTimeZoneAlternative(calendarEvent: VEvent): TimeZone =
+    CalendarTimeZoneField.getTimeZoneFromTZID(calendarEvent)
+      .getOrElse(TimeZone.getTimeZone(ZoneId.of("UTC")))
+}
+
+case class CalendarStartField(value: ZonedDateTime) extends AnyVal
+
+object CalendarEndField {
+  def from(calendarEvent: VEvent): Option[CalendarEndField] =
+    Option(calendarEvent.getEndDate(false))
```

Comment:
```suggestion
    Option(calendarEvent.getEndDate(true))
```
?
![image](https://user-images.githubusercontent.com/55171818/223653982-ddf42c64-d562-4cfb-af62-967b01182c99.png)
And maybe a test for it? (DTSTART and DURATION is specified while DTEND is missing)

Reply:
Which `end` value will have higher priority?
(1) from `DTEND` property
(2) based on `DTSTART` + `DURATION` 

IMO it should be (1), then -> `false`

Reply:
The way I understand it, DTEND has always high priority? The boolean is just to say if you want, or not, use DTSTART and DURATION to calculate DTEND `IF` DTEND is missing. But if DTEND is present, it will always return DTEND.

I think Quan's comment makes sense?

Reply:
tests can confirm this :p